sound/soc/sof/pcm.c

```diff
@@ -302,6 +287,14 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 	dev_dbg(component->dev, "pcm: trigger stream %d dir %d cmd %d\n",
 		spcm->pcm.pcm_id, substream->stream, cmd);
 
+	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_STOP:
+		break;
+	default:
+		spcm->stream_stopped[substream->stream] = false;
+		break;
+	}
```

Comment:
Just unconditionally clear the flag?

Reply:
or handle the flag in one place:
```
	/*
	 * The stream is not stopped in case of platform_stop_during_hw_free == true
	 * During consequent prepare callback it will be needed to be stopped before
	 * starting the stream again (part of xrun handling)
	 */
	if (cmd == SNDRV_PCM_TRIGGER_STOP && pcm_ops &&
	    pcm_ops->platform_stop_during_hw_free)
		spcm->trigger_stop_pending[substream->stream] = true;
	else
		spcm->trigger_stop_pending[substream->stream] = false;
```
but after the second switch() section?